cmd/ebrelayer/txs/relayToCosmos.go

```diff
@@ -13,27 +13,32 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/x/auth/client/utils"
 	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"go.uber.org/zap"
 )
 
 var (
 	nextSequenceNumber uint64 = 0
+	errorMessageKey           = "errorMessage"
 )
 
 // RelayToCosmos applies validator's signature to an EthBridgeClaim message containing
 // information about an event on the Ethereum blockchain before relaying to the Bridge
 func RelayToCosmos(cdc *codec.Codec, moniker, password string, claims []types.EthBridgeClaim, cliCtx context.CLIContext,
-	txBldr authtypes.TxBuilder) error {
+	txBldr authtypes.TxBuilder, sugaredLogger *zap.SugaredLogger) error {
 	var messages []sdk.Msg
-	log.Printf("ebrelayer RelayToCosmos with %d claims\n", len(claims))
-	log.Printf("ebrelayer RelayToCosmos nextSequenceNumber is %d\n", nextSequenceNumber)
+
+	sugaredLogger.Infow("relay prophecies to cosmos.",
+		"claim amount", len(claims),
+		"next sequence number", nextSequenceNumber)
 
 	for _, claim := range claims {
 		// Packages the claim as a Tendermint message
 		msg := ethbridge.NewMsgCreateEthBridgeClaim(claim)
 
 		err := msg.ValidateBasic()
 		if err != nil {
-			log.Println("failed to get message from claim with:", err.Error())
+			sugaredLogger.Errorw("failed to get message from claim.",
```

Comment:
How does this print out when you print the errorMessageKey and then you print the error afterwards?

Do errorMessageKey and error get concatenated into a single value and then `failed to get message from claim.` is the key to the object that is printed out?

Reply:
Seems like my answer is here https://pkg.go.dev/go.uber.org/zap#SugaredLogger.Errorw

There is a string, then keys and values are passed as variadic args so this should print correctly.